packages/venia-ui/lib/components/FilterSidebar/filterSidebar.js

```diff
@@ -0,0 +1,145 @@
+import React, { useMemo, useCallback, useRef, Fragment } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { array, arrayOf, shape, string, number } from 'prop-types';
+import { useFilterModal } from '@magento/peregrine/lib/talons/FilterModal';
+
+import { mergeClasses } from '../../classify';
+import LinkButton from '../LinkButton';
+import CurrentFilters from '../FilterModal/CurrentFilters';
+import FilterBlock from '../FilterModal/filterBlock';
+import defaultClasses from './filterSidebar.css';
+
+const DEFAULT_FILTERS_OPEN_COUNT = 3;
+
+/**
+ * A view that displays applicable and applied filters.
+ *
+ * @param {Object} props.filters - filters to display
+ */
+const FilterSidebar = props => {
+    const { filters, filtersOpen } = props;
+    const talonProps = useFilterModal({ filters });
```

Comment:
@sirugh It hurts to duplicate that much code, but I'll do it for you ;). The state should hopefully cross over since it's essentially based off of the URL, but I'll test to be sure.

The work is slightly based off of one of our projects and their approach was to use the same component. However there were too many "if" conditions for it to be maintainable long term, so we went for the approach of having a separate component.

Reply:
> It hurts to duplicate that much code

I know :/ but the alternative is bug surface (for now). Eventually we will have a pattern for duplicated logic for talons, but we don't want to solve that here :)